adoptingerlang.org

```diff
@@ -4101,20 +4101,118 @@ With these broad lines covered, we can start exploring the processes and practic
+- we're already doing polyglot and there's this one component nobody wants to touch because a dev once wrote it in a language nobody else knows
+
+Build engineering is a legit concern, in the same way many organisations with a dedicated IT department ship out a single type of computers for every developer to reduce the surface area of what is required. That being said, this is a less prevalent issue with the advent of pervasive containerization. Docker images and runtimes such as Kubernetes already provide a one-size-fits-all interface that extends from local development to production runs. Maintaining build images with the help of people familiar with the programming languages on board mostly reduces the complexity of everything involved over time.
+
+There isn't necessarily a great way to work around dependency explosion and monitoring, short of using tools that scan and report issues for you. That being said, dependency surface area is mostly a problem of managing complexity, and isn't necessarily tied to how many languages you used. For example, we've seen single Javascript demo projects with dependency trees larger than established Erlang software used for entire departments in a company. Other tools, such as dependency vulnerability scanners, are more and more adding support for Hex packages.
+
+If your company has a procedure where each dependency must be vetted by the a security team, the legal department (for licensing purposes), and engineers (for code quality), you'll find out that in practice, it doesn't work great. People will either stick to a pre-whitelisted list of libraries or will only review top-level dependencies. In some cases, they'll never update outdated libraries because they prefer to avoid all the red tape. In other cases the developers will just write the dependency into their project in order to avoid having to cross many departments of reviews if they feel they gotta ship things fast, and you'll end up with code that barely gets reviewed at all, and possibly of poor quality because they may have rewritten established libraries. We consider this to be an organisational issue, not a programming language issue.
+
+Uncool languages or languages that are "hard to hire for" can always be a problem. Unfortunately, this is a problem you can _never_ actually solve, because nobody knows the future. Is it worse to have a single main language people stop liking and then none of your stack is easy to staff, or to have only half of it being hard to hire for? What about cases like Python, where the migration from 2.x to 3.x has been so hard that libraries and components essentially consider them to be distinct languages? We consider that organisations that do not marry themselves to a single language but instead make a habit of switching between a few of them (and not just a ton of them) train themselves to better adjust to variations in the market of programming languages.
```

Comment:
```suggestion
Uncool languages or languages that are "hard to hire for" will always be a problem. Unfortunately, this isn't a problem you can  actually solve, because no one knows the future.  Is it worse to have a single main language people stop liking and then none of your stack is easy to staff, or to have only half of it being hard to hire for? What about cases like Python, where the migration from 2.x to 3.x has been so hard that libraries and components essentially consider them to be distinct languages? We consider that organisations that do not marry themselves to a single language but instead make a habit of switching between a few of them (and not just a ton of them) train themselves to better adjust to variations in the market of programming languages.
```